### ISSUE TITLE
[Feature] Add description to card to avoid needing to make a separate note for added details

### ISSUE THREAD

**Issue by nickjj** [mgmeyers/obsidian-kanban#94]
Hi,

First up, really nice plugin, I just started using Obsidian and came across yours for making a Kanban board.

One thing that I found myself doing a lot with another Kanban board solution was adding both a title and description to a specific card. But with this plugin and Obsidian at the moment it looks like adding a separate note to a card is the only way to add more details about a card but this leads to a number of non-ideal situations:

For example, I had a card title with `Verify webhooks without extra API call` and I wanted to add more context around this by adding a description of: 

```
You can check the signature of the webhook in the HTTP header instead of having to look up the event afterwards as a separate API call.

Details: https://stripe.com/docs/webhooks/signatures#verify-manually
```

- I made a new note from this card but now it dumped the note in the root of my vault, except my Kanban board is a few directories deep so it took a bunch of clicks and drags to move it there.

- Then I decided I really wanted to rename that card with `Verify Stripe webhooks without extra API call` so I renamed the card, except the note file associated to it still has the old name so I now I need to manually rename the note file or it's out of sync with the card name.

- Then I added 15 cards to 1 column and started moving them across the Kanban board but now I have this single directory of files sitting there and it's hard to tell which state those files are in and trying to micro manage them by creating sub-directories for each list name and moving them there feels like getting bogged down in the details.

Perhaps all of these problems could be solved by supporting a new description field that you can add to a card so you can write your own notes for a card without needing an explicit note file.

Then you have the option of treating this more like daily notes where you keep your card titles short, add sprinkles of a description to it for extra context and link out to dedicated notes that are unrelated to the Kanban board itself.

What do you think?

**Comment by skillhacker-code** [comment]
Yes it would be great if this feature is there

**Comment by pjeby** [comment]
To address a few items from above:

* If you set either the vault or kanban options to create the notes alongside the board, they will be created there instead of your vault root
* If you rename the note instead of the card (right click the link and pick "Rename") then both the card and the note will be renamed  (Obsidian in general lets you rename notes and update all links to them, including links in Kanban boards)
* If you use the Obsidian backlinks view while looking at one of the notes, it will show you (when you ctrl/cmd-hover preview) exactly where it is in your kanban board.  (You'll need to enable Page Preview and Backlinks in the "Core Plugins" page.)
* You can link to existing notes from a Kanban board, just type `[[` in a card to bring up suggestions
* You can have more description in a card by typing it outside the link -- you can actually add as much text to a card as you want, as long as it's all one markdown "paragraph"
* You can have even more description and text by using the new metadata feature that pulls front matter fields from the first note linked in the card
* Oh yeah -- one card can link to as many notes as you want, and can link to headings or blocks within the target notes, not just a whole note.

tl;dr: converting a card to a note is just a shortcut for putting `[[ ]]` around the note text and clicking the link to create the note, but card descriptions can be any markdown paragraph, and they can include formatting as well as internal and external links

So you could make a card with this text:

```markdown
[You can check the signature of the webhook in the HTTP header instead of having to look up the event afterwards as a separate API call.](https://stripe.com/docs/webhooks/signatures#verify-manually)
```

And it would be rendered thus:

[You can check the signature of the webhook in the HTTP header instead of having to look up the event afterwards as a separate API call.](https://stripe.com/docs/webhooks/signatures#verify-manually)

With a working link.

**Comment by nickjj** [comment]
Thanks for the thorough reply, I really appreciate it. I'll give some of those patterns a go.

**Comment by mgmeyers** [comment]
FYI, 1.0.0 allows multi-line cards (shift+enter by default to create a new line). This doesn't allow hidden descriptions, but you can add any markdown you want to a card now.

**Comment by firinael** [comment]
This'd be very good if implemented together with nesting.

**Comment by skillhacker-code** [comment]
Thank You so much Mg Meyers for adding option for multi line card. It is very useful . 

Thank you so much
